### PR TITLE
GET POST AND DELETE Reminder API Created

### LINKED ISSUE #36: create reminder endpoints
Create endpoints for the reminder functionality.

1. `GET` endpoint to get all set reminders for the patients.
2. `PATCH` endpoint to update field(s) of a reminder.
3. `POST` endpoint to create a new reminder.
4. `DELETE` endpoint to remove an existing reminder.

Make sure the endpoints are tested.

### PR DESCRIPTION
**Pull request by alian1001**
EMAIL IS USED FOR
-Reminder update, add, delete
-Reminder when time is triggered
-This clinician has invited you to connect, if user not registered, send invitation token, otherwise add to contacts book

POST

{
  "setById": "user_2kb0nnXRX4SdwJAO5gVMyF12Xmw",
  "setForId": "user_2kit0iBGwimRaKgg3WjIIqZPAvG",
  "scheduledAt": "2024-08-20T10:00:00Z",
  "startDate": "2024-08-19T00:00:00Z",
  "endDate": "2024-09-19T00:00:00Z",
  "description": "Appointment reminder",
  "type": "APPOINTMENT",
  "intervalInDays": 30
}

PUT

{
  "id": 1,
  "setById": "user_2kb0nnXRX4SdwJAO5gVMyF12Xmw",
  "setForId": "user_2kit0iBGwimRaKgg3WjIIqZPAvG",
  "scheduledAt": "2024-08-21T10:00:00Z",
  "startDate": "2024-08-19T00:00:00Z",
  "endDate": "2024-09-20T00:00:00Z",
  "description": "Updated appointment reminder",
  "type": "APPOINTMENT",
  "intervalInDays": 60,
  "isReminderActive": true
}

DELETE

{
  "id": 1
}

Do let me know if anything else has gone wrong.

### CODE REVIEW

File: src/app/api/reminder/route.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import { PrismaClient } from "@prisma/client";
+import { HttpStatusCode } from "axios";
+import { Resend } from "resend";
+import { EmailTemplate } from "../../components/emailTemplate/email-template";
+
+const prisma = new PrismaClient();
+const resend = new Resend(process.env.RESEND_API_KEY);
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: Update the env template `.env.example` to include this key.

File: src/app/api/reminder/route.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+// Helper function to send an email via Resend
+async function sendEmail(
+  subject: string,
+  actionType: "created" | "updated" | "deleted",
+  templateData: any
+) {
+  try {
+    const { data, error } = await resend.emails.send({
+      from: "OrthoChat <onboarding@resend.dev>",
+      to: ["alian.haidar01@gmail.com"], // Hardcoded email recipient
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: accept recipient email as param

File: src/app/api/email/route.ts
Comment (ellenxxxiao) [inline src/app/api/email/route.ts:1]: Is this a test route? If yes you can remove it

File: src/app/login/page.tsx
Comment (ellenxxxiao) [inline src/app/login/page.tsx:1]: Same here, remove code for testing purposes when submitting PR

File: src/app/components/emailTemplate/email-template.tsx
Comment (ellenxxxiao) [inline src/app/components/emailTemplate/email-template.tsx:1]: Try using Tailwind CSS classes.
Could also make use of React Email to use some components. Refer to https://resend.com/docs/dashboard/emails/email-templates

File: src/app/api/reminder/route.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  } catch (error) {
+    return new Response(JSON.stringify({ error: getErrorMessage(error) }), {
+      status: HttpStatusCode.InternalServerError,
+      headers: { "Content-Type": "application/json" }
+    });
+  }
+}
+
+// PUT: Update an existing reminder
+export async function PUT(request: Request) {
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: Use `PATCH`

File: src/app/api/reminder/route.ts [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+// GET: Fetch reminders
+export async function GET(request: Request) {
+  try {
+    const url = new URL(request.url);
+    const setById = url.searchParams.get("setById");
+    const setForId = url.searchParams.get("setForId");
+    const isReminderActive =
+      url.searchParams.get("isReminderActive") === "true";
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: Could be simplified by:

```typescript
import { NextRequest, NextResponse } from 'next/server';

export async function GET(
  request: NextRequest,
) {
  //...

  // get search params
  const searchParams = request.nextUrl.searchParams;
  const isActive = searchParams.get("isActive");

  //...
}

```

Refer to https://nextjs.org/docs/app/api-reference/functions/next-request#nexturl

File: src/app/api/reminder/route.ts [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+        headers: { "Content-Type": "application/json" }
+      }
+    );
+  } catch (error) {
+    return new Response(JSON.stringify({ error: getErrorMessage(error) }), {
+      status: HttpStatusCode.InternalServerError,
+      headers: { "Content-Type": "application/json" }
+    });
+  }
+}
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: `PATCH` and `DELETE` operations should be performed on a specific reminder only, so these two endpoints should in file `api/reminder/[id]/route.ts`. The param `id` refers to the reminder ID. Refer to the `/users` endpoint.

File: src/app/api/reminder/route.ts [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+  try {
+    const {
+      setById,
+      setForId,
+      scheduledAt,
+      startDate,
+      endDate,
+      description,
+      type,
+      intervalInDays
```
Comment (ellenxxxiao) [inline src/app/api/reminder/route.ts]: Could be converted into a type

**Review by ellenxxxiao — CHANGES_REQUESTED** [review]
Nice work, refer to comments

File: src/app/api/reminder/route.ts (hunk 2 above)
Comment (alian1001) [reply src/app/api/reminder/route.ts]: no domain to be used. stick with hardcoded for now.

**Pull request merged**